nntrainer/src/databuffer.cpp

```diff
@@ -187,12 +185,15 @@ bool DataBuffer::getDataFromBuffer(BufferType type, vec_4d &outVec,
   switch (type) {
   case BUF_TRAIN: {
     std::vector<int> list;
-    std::unique_lock<std::mutex> ultrain(readyTrainData);
-    cv_train.wait(ultrain, [this]() -> int { return trainReadyFlag; });
-
-    if (train_data.size() < batch_size || trainReadyFlag == DATA_ERROR ||
-        trainReadyFlag == DATA_END) {
-      return false;
+    while (true) {
+      std::unique_lock<std::mutex> ultrain(readyTrainData);
+      cv_train.wait(ultrain, [this]() -> int { return trainReadyFlag; });
+      if (trainReadyFlag == DATA_ERROR || trainReadyFlag == DATA_END) {
```

Comment:
```suggestion
      if (trainReadyFlag == DATA_ERROR || trainReadyFlag == DATA_END || train_data.empy()) {
```

to eliminate edge cases? (and for other cases)

Reply:
it will be checked at line 194